packages/framework-provider-aws/src/library/events-adapter.ts

```diff
@@ -110,3 +110,55 @@ export async function storeEvents(
   await dynamoDB.batchWrite(params).promise()
   logger.debug('[EventsAdapter#storeEvents] EventEnvelope stored')
 }
+
+export async function destroyEntity(
+  dynamoDB: DynamoDB.DocumentClient,
+  config: BoosterConfig,
+  logger: Logger,
+  entityTypeName: string,
+  entityID: UUID
+): Promise<void> {
+  logger.debug('[EventsAdapter#destroyEntity] Destroying entity')
+
+  // We need to fetch the entity events and snapshots
+  const events = await readEntityEventsSince(dynamoDB, config, logger, entityTypeName, entityID)
+  const snapshots = await readEntityEventsSince(
+    dynamoDB,
+    config,
+    logger,
+    entityTypeName,
+    entityID,
+    originOfTime,
+    'snapshot'
+  )
+
+  if (events) {
+    let eventsToDelete = [...events]
+    if (snapshots) {
+      eventsToDelete = [...eventsToDelete, ...snapshots]
+    } else {
+      logger.debug(`[EventsAdapter#destroyEntity] No snapshots found for entity ${entityTypeName} with id: ${entityID}`)
+    }
+
+    const params: DynamoDB.DocumentClient.BatchWriteItemInput = {
+      RequestItems: {
+        [config.resourceNames.eventsStore]: eventsToDelete.map((eventEnvelope) => ({
+          DeleteRequest: {
+            Key: {
+              [eventsStoreAttributes.partitionKey]: partitionKeyForEvent(
+                eventEnvelope.entityTypeName,
+                eventEnvelope.entityID,
+                eventEnvelope.kind
+              ),
+              [eventsStoreAttributes.sortKey]: eventEnvelope.createdAt,
+            },
+          },
+        })),
+      },
+    }
+    await dynamoDB.batchWrite(params).promise()
```

Comment:
For the future we should create chunks of 25 items to be written. The [batchWriteItem](https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_BatchWriteItem.html) operation has that limit.

Reply:
Can you add a `TODO` comment about this in this place, please?